src/sys_process_lifecycle.c

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright (C) 2015 The University of Notre Dame
+This software is distributed under the GNU General Public License.
+See the file LICENSE for details.
+*/
+
+#include "console.h"
+#include "process.h"
+#include "iso.h"
+#include "memorylayout.h" // PROCESS_ENTRY_POINT
+#include "sys_process_lifecycle.h"
+
+uint32_t sys_exit(uint32_t code) {
+    process_exit(code);
+    return 0;
+}
+
+uint32_t sys_yield() {
+    process_yield();
+    return 0;
+}
+
+uint32_t sys_run(char *process_path, struct process_permissions *child_permissions) {
+    if (child_permissions == 0) {
+        return -1;
+    }
+
+    // Load process data
+    struct iso_dir *root_dir = iso_dopen("/", 3);
+    if (root_dir == 0) {
+        return -1;
+    }
+
+    struct iso_file *proc_file = iso_fopen(process_path, root_dir->ata_unit);
+    if (proc_file == 0) {
+        return -1;
+    }
+
+    uint8_t *process_data = kmalloc(proc_file->data_length);
+    if (process_data == 0) {
+        return -1;
+    }
+
+    int num_read = iso_fread(process_data, proc_file->data_length, 1, proc_file);
+    if (num_read == 0) {
+        return -1;
+    }
+
+    // store the current number of pages used, so we can see how many the child uses
+    int page_count_before_child = current->number_of_pages_using;
+
+    // Create a new process(page, page)
+    struct process *child_proc = process_create(PAGE_SIZE, PAGE_SIZE);
+    child_proc->permissions = *child_permissions; // store as value, but we needed to pass as reference because syscall
+    child_proc->parent = current; // store the child's parent
```

Comment:
Great thought; I do believe it's possible. Do we want to make this function atomic? Probably not. I'll update to pass the parent process in that inner function.